src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+            updateStatement.setObject(2, car.getManufacturer().getId());
+            updateStatement.setObject(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't update car: " + car.getModel()
+                    + " from cars table. ",
+                    throwable);
+        }
+        removeRelation(car);
+        insertDriver(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement
+                        = connection.prepareStatement(deleteQuery)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() != 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete a cars by id " + id + " ",
+                    throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id as car_id, c.name as car_name, m.name as manufacturer_name, "
+                + "m.id as manufacturer_id,"
+                + " country as manufacturer_country FROM cars c JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id WHERE c.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCars = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCars.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarWithManufacturer(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get all cars from DB by id ",
+                    throwable);
+        }
+        if (cars.size() != 0) {
+            for (Car car : cars) {
+                car.setDriver(getDriverForCar(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    private Car getCars(ResultSet resultSet) throws SQLException {
+        Long newId = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Manufacturer manufacturer = new Manufacturer();
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car();
+        car.setId(newId);
+        car.setModel(model);
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private List<Driver> getDriverForCar(Long driverId) {
+        String getAllDriverForCarRequest = "SELECT id, name, country FROM driver d"
```

Comment:
how does it work if neither cars, nor drivers table don't have a country column?
```suggestion
        String getAllDriverForCarRequest = "SELECT id, name, country FROM drivers d"
```